actionpack/test/dispatch/cookies_test.rb

```diff
@@ -408,6 +418,30 @@ def test_setting_cookie_with_strict
     assert_equal({ "user_name" => "david" }, @response.cookies)
   end
 
+  def test_setting_cookie_with_false
+    @request.env["action_dispatch.cookies_same_site_protection"] = proc { false }
```

Comment:
Can we test with `nil` (instead of `false`), and perhaps elaborate the test name?  I realize this is mimicking the name of the test above it (`test_setting_cookie_with_strict`), but that test name was a little cryptic to start with.

```suggestion
  def test_setting_cookie_with_same_site_nil
    @request.env["action_dispatch.cookies_same_site_protection"] = proc { nil }
```